back_end/src/models/major.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from typing import List
+
+from ..setup import db
+
+from flask_login import UserMixin
+
+
+
+class Major(db.Model):
+    __tablename__ = "Majors"
+
+    id = db.Column(db.Integer, primary_key=True)
+    major_code = db.Column(db.String(255), unique=True, nullable=False)
+    title = db.Column(db.String(255), unique=True, nullable=False)
+    degree_type = db.Column(db.String(255), unique=True, nullable=False)
+
+    def __init__(self, **kwargs):
+        super(Major, self).__init__(**kwargs)
+
+    def to_json(self):
+        ret = {}
+        ret['id'] = self.id
+        ret['major_code'] = self.major_code
+        ret['title'] = self.title
+        ret['degree_type'] = self.degree_type
+        return ret
+
+    def update_attr(self, major_code: str, title: str,
+                    degree_type: str) -> bool:
+        if major_code:
+            self.major_code = major_code
+        if title:
+            self.title = title
+        if degree_type:
+            self.degree_type = degree_type
+        return True
+
+    def save(self):
+        db.session.commit()
+
+    @staticmethod
+    def create_major(major_code: str, title: str, degree_type: str) -> bool:
+        major = Major(major_code=major_code, title=title, degree_type=degree_type)
+        db.session.add(major)
+        major.save()
+        return True
+
+    @staticmethod
+    def get_majors() -> List[Major]:
+        majors = Major.query.all()
+        majors = list(map(lambda x: x.to_json(), majors))
+        return majors
+
+    @staticmethod
+    def get_major(major_id: int) -> Major:
+        return Major.query.filter_by(id=major_id).first()
```

Comment:
This is an inconsistency. in the above method, you are returning a list of objects that are already jsonified into a dict. However, in this method, you are only returning the `Major` object. If that is the case, in the `api` it might be potential problems. So maybe we can just return a list of objects in the above method.